spec/unit/models/services/route_binding_spec.rb

```diff
@@ -52,5 +52,150 @@ module VCAP::CloudController
         expect(binding.errors[:service_instance]).to eq [:space_mismatch]
       end
     end
+
+    describe '#save_with_new_operation' do
+      let(:space) { Space.make }
+      let(:service_offering) { Service.make(requires: ['route_forwarding']) }
+      let(:service_plan) { ServicePlan.make(service: service_offering) }
+      let(:service_instance) { ManagedServiceInstance.make(space: space, service_plan: service_plan) }
+      let(:route) { Route.make(space: space) }
+      let(:route_service_url) { 'https://foo.com' }
+      let(:route_binding) do
+        RouteBinding.new(
+          service_instance: service_instance,
+          route: route,
+        )
+      end
+
+      it 'updates attributes and creates a new last_operation object' do
+        last_operation = {
+          state: 'in progress',
+          type: 'create',
+          description: '10%'
+        }
+        attributes = {
+          route_service_url: route_service_url
+        }
+        result = route_binding.save_with_new_operation(attributes, last_operation)
+
+        expect(result).to eq(route_binding)
+        expect(route_binding.service_instance).to eq(service_instance)
+        expect(route_binding.route).to eq(route)
+        expect(route_binding.route_service_url).to eq(route_service_url)
+        expect(route_binding.last_operation.state).to eq 'in progress'
+        expect(route_binding.last_operation.description).to eq '10%'
+        expect(route_binding.last_operation.type).to eq 'create'
+        expect(RouteBinding.count).to eq(1)
+      end
+
+      context 'when saving the binding operation fails' do
+        before do
+          allow(RouteBindingOperation).to receive(:create).and_raise(Sequel::DatabaseError, 'failed to create new-binding operation')
+        end
+
+        it 'should rollback the binding' do
+          expect { route_binding.save_with_new_operation({}, { state: 'will fail' }) }.to raise_error(Sequel::DatabaseError)
+          expect(RouteBinding.count).to eq(0)
+        end
```

Comment:
Not the most egregious and probably something we do in many tests in the codebase, but stubbing `create` leaks quite a bit of implementation detail (for instance, now you can't refactor to `#.new` -> `.save` if you needed to for whatever reason).

It's a bit more leaking the business / DB constraints, but I'd probably try and set the operation up with data that will actually cause an error to be raised.